File: app/src/main/java/th/co/thekhaeng/materialstyletemplate/adapter/base/BaseViewHolder.java
package th.co.thekhaeng.materialstyletemplate.adapter.base;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by Nonthawit on 7/30/2016.
 */

public abstract class BaseViewHolder extends RecyclerView.ViewHolder{

    public BaseViewHolder( ViewGroup parent, int layout ){
        super( LayoutInflater.from( parent.getContext() ).inflate( layout, parent, false ) );
        bindView( itemView );
    }


    protected abstract void bindView( View view );

    public Context getContext(){
        return itemView.getContext();
    }

}

File: app/src/main/kotlin/com/thekhaeng/materialstyletemplate/adapter/base/BaseViewHolder.kt
package com.thekhaeng.materialstyletemplate.adapter.base

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

/**
 * Created by Nonthawit on 7/30/2016.
 */

abstract class BaseViewHolder(parent: ViewGroup, layout: Int)
    : RecyclerView.ViewHolder(
        LayoutInflater
                .from(parent.context)
                .inflate(layout, parent, false)) {

    val context: Context
        get() = itemView.context

    init {
        setupView()
    }


    open fun setupView() {}

}
